### PR TITLE
Add index_errors association matcher

### PR DESCRIPTION
**Pull request by brendanthomas1**
Hello! Our team was adding `index_errors` to a couple `has_many` associations and we noticed that there's no matcher for that yet. It looked pretty similar to `autosave` since it's just a boolean, so I pretty much copied that 😄

### CODE REVIEW

File: lib/shoulda/matchers/active_record/association_matcher.rb [hunk 1]
```diff
@@ -1257,6 +1282,22 @@ def autosave_correct?
           end
         end
 
+        def index_errors_correct?
```
[the remark this answers is not in the document]
Comment (brendanthomas1) [reply lib/shoulda/matchers/active_record/association_matcher.rb]: This should be down to 10 lines now 🙂

**Comment by mcmire** [comment]
I've never seen the `index_errors` option before. What does it do?

**Comment by mcmire** [comment]
Oh very interesting, okay. Thanks for the clarification!

**Comment by mcmire** [comment]
Thanks! I've tweaked this a bit for style and merged this as 795ca688bff08590dbd2ab6f2b51ea415e0c7473.

**Pull request closed without merging**